### PR TITLE
Add API properties to an API

### PR DESCRIPTION
**Pull request by thisaltennakoon**
$subject
Related PR https://github.com/wso2/apk/pull/1251

**Config for Adding APIProperties to an API**
```
spec:
  apiProperties:
    - name: propertyName1
      value: "propertyValue1"
    - name: propertyName2
      value: "propertyValue2"
```

**Added APIProperties can be accessed inside interceptor services in both Request and Response paths**
```
invocationContext: {
   apiProperty: "{'propertyName1':'propertyValue1','propertyName2':'propertyValue2'}"
}
```

### CODE REVIEW

File: samples/request-response-mediation-interceptors/k8s-resources/api.yaml [hunk 1]
```diff
@@ -24,6 +24,11 @@ spec:
   apiType: REST
   apiVersion: 1.0.0
   context: /interceptor/1.0.0
+  apiProperties:
```
Comment (BLasan) [inline samples/request-response-mediation-interceptors/k8s-resources/api.yaml:27]: Use singular ones similar to the `env` defined in K8s. Refer https://kubernetes.io/docs/tasks/inject-data-application/define-environment-variable-container/

File: adapter/internal/operator/apis/dp/v1alpha1/api_types.go [hunk 2]
```diff
@@ -89,6 +89,14 @@ type APISpec struct {
 	//
 	// +optional
 	SystemAPI bool `json:"systemAPI"`
+
+	APIProperty []Property `json:"apiProperties,omitempty"`
```
Comment (BLasan) [inline adapter/internal/operator/apis/dp/v1alpha1/api_types.go:93]: Here too

**Review by BLasan — CHANGES_REQUESTED** [review]

File: samples/request-response-mediation-interceptors/k8s-resources/api.yaml (hunk 1 above)
Comment (AmaliMatharaarachchi) [reply samples/request-response-mediation-interceptors/k8s-resources/api.yaml:27]: `apiProperties` looks fine to me

File: adapter/internal/operator/apis/dp/v1alpha1/api_types.go (hunk 2 above)
Comment (AmaliMatharaarachchi) [inline adapter/internal/operator/apis/dp/v1alpha1/api_types.go:93]:
```suggestion
	APIProperties []Property `json:"apiProperties,omitempty"`
```

**Review by AmaliMatharaarachchi — APPROVED** [review]

**Pull request merged**